providers/flagd/pkg/provider.go

```diff
@@ -118,6 +132,27 @@ func WithHost(host string) ProviderOption {
 	}
 }
 
+// WithoutCache disables caching
+func WithoutCache() ProviderOption {
+	return func(p *Provider) {
+		p.Cache = nil
+	}
+}
+
+// WithCustomCache applies a custom cache implementation
+func WithCustomCache(cacheImplementation Cache) ProviderOption {
+	return func(p *Provider) {
+		p.Cache = cacheImplementation
+	}
+}
+
+// WithContext supplies the given context to the event stream
+func WithContext(ctx context.Context) ProviderOption {
```

Comment:
I've opted to allow context to be included via an option rather than the idiomatic first function parameter for the sake of not being a breaking change.

Reply:
I like this approach